ios-sample-app/ios-sample-app/Controllers/IdentityController.m

```diff
@@ -18,25 +22,162 @@ @implementation IdentityController
+        // This can also be called before SDK init if you want the first session to include the Custom User Id.
+        [Singular setCustomUserId:customUserId];
+        
+        // Logging for Testing
+        NSLog(@"Custom User ID Set %@", customUserId);
+        [Utils displayMessage:[[NSString alloc] initWithFormat:@"Custom User Id set to: %@", customUserId] withView:self];
     }
-    
-    // Once set, the Custom User Id will persist between runs until `[Singular unsetCustomUserId]` is called.
-    // This can also be called before SDK init if you want the first session to include the Custom User Id.
-    [Singular setCustomUserId:customUserId];
-
-    [Utils displayMessage:[[NSString alloc] initWithFormat:@"Custom User Id set to: %@", customUserId] withView:self];
 }
 
 - (IBAction)unsetCustomUserId:(id)sender {
     [Singular unsetCustomUserId];
+    
+    // Logging for Testing
+    NSLog(@"Custom User ID Unset");
     [Utils displayMessage:@"Custom User Id unset" withView:self];
 }
 
+- (void)requestTrackingAuthorization {
+    AppDelegate *DataDelegate = (AppDelegate *)[[UIApplication sharedApplication] delegate];
+    // call requestTrackingAuthorizationWithCompletionHandler from ATTrackingManager to start the user consent process
+    if (@available(iOS 14, *)) {
+        [ATTrackingManager requestTrackingAuthorizationWithCompletionHandler:^(ATTrackingManagerAuthorizationStatus status) {
+            // Tracking authorization completed. Start loading ads here.
+            switch (status) {
+                case 0: self.att_state = @"(0) Not Determined";
+                case 1: self.att_state = @"(1) Restricted";
+                case 2: self.att_state = @"(2) Denied";
+                case 3: self.att_state = @"(3) Authorized";
+                }
+            self.s_idfa = [[[ASIdentifierManager sharedManager] advertisingIdentifier] UUIDString];
+            self.s_idfv = [[[UIDevice currentDevice] identifierForVendor] UUIDString];
+            [self.idfa_value setText:self.s_idfa];
+            [self.idfv_value setText:self.s_idfv];
+            [self.att_value setText:self.att_state];
+            DataDelegate.sharedMessage = [NSString stringWithFormat:@"Sample App Device Info:\n\nAdvertising ID (IDFA): %@ \n\nIDFV: %@",self.s_idfa,self.s_idfv];
+          }];
+    } else {
+        [self.idfa_value setText:[[[ASIdentifierManager sharedManager] advertisingIdentifier] UUIDString]];
+        [self.idfv_value setText:[[[UIDevice currentDevice] identifierForVendor] UUIDString]];
+        [self.att_value setText:@"(0) Not Determined < iOS14"];
+    }
+    
+    // Logging for Testing
+    NSLog(@"IDFA: %@", self.s_idfa);
+    NSLog(@"IDFV: %@", self.s_idfv);
+    NSLog(@"ATT Status: %@", self.att_value);
+}
+
+- (IBAction)shareText:(id)sender {
+    // create a share message
+    AppDelegate *DataDelegate = (AppDelegate *)[[UIApplication sharedApplication] delegate];
+    NSString* sharedText = DataDelegate.sharedMessage;
```

Comment:
can we have sharedMessage as IdentityController variable?

Reply:
Updated to use ViewController variable.